### PR TITLE
initial alert controller structure

### PR DESCRIPTION
**Pull request by mphammer**

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/742#" title="Author self-approved">mphammer</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: meta-builder/config/crd/bases/synopsys.com_alerts.yaml [hunk 1]
```diff
@@ -30,15 +30,77 @@ spec:
         spec:
           description: AlertSpec defines the desired state of Alert
           properties:
-            finalYamlUrl:
-              description: 'INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-                Important: Run "make" to regenerate code after modifying this file'
+            EncryptionGlobalSalt:
```
Comment (yashbhutwala) [inline meta-builder/config/crd/bases/synopsys.com_alerts.yaml]: s/EncryptionGlobalSalt/encryptionGlobalSalt

File: meta-builder/config/crd/bases/synopsys.com_alerts.yaml [hunk 2]
```diff
@@ -30,15 +30,77 @@ spec:
...
               type: string
-            shaOfFinalYaml:
+            EncryptionPassword:
```
Comment (yashbhutwala) [inline meta-builder/config/crd/bases/synopsys.com_alerts.yaml]: s/EncryptionPassword/encryptionPassword

File: meta-builder/config/crd/bases/synopsys.com_alerts.yaml [hunk 3]
```diff
@@ -30,15 +30,77 @@ spec:
...
+                type: string
+              type: array
+            exposeService:
+              type: string
+            imageRegistries:
+              items:
+                type: string
+              type: array
+            namespace:
+              description: 'Important: Run "make" to regenerate code after modifying
```
Comment (yashbhutwala) [inline meta-builder/config/crd/bases/synopsys.com_alerts.yaml]: I think annotations produced this by accident

File: meta-builder/controllers/alert_controller.go [hunk 4]
```diff
@@ -65,10 +68,17 @@ func (r *AlertReconciler) GetRuntimeObjects(cr interface{}) (map[string]runtime.
 	// 1. Get List of Runtime Objects (Base Yamls)
 	// TODO: either read contents of yaml from locally mounted file
 	// read content of full desired yaml from externally hosted file
-	byteArrayContentFromFile, err := controllers_utils.HttpGet(alert.Spec.FinalYamlUrl)
+	FinalYamlUrl := "https://raw.githubusercontent.com/mphammer/customer-on-prem-alert-final-yaml/master/base-on-prem-alert-final.yaml"
```
Comment (yashbhutwala) [inline meta-builder/controllers/alert_controller.go]: Should this now be relative path to the above file: alert_runtime_objects.yaml

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/742#" title="Author self-approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/742#issuecomment-520488133" title="LGTM">yashbhutwala</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer,yashbhutwala]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/742#" title="Author self-approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/742#pullrequestreview-273793798" title="Approved">yashbhutwala</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer,yashbhutwala]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**